### PR TITLE
Add search field reset button

### PR DESCRIPTION
**Pull request by MKodde**
It is shown once the user starts typing text into the search bar. When the search term is cleared, the poirot icon is shown again. This toggle is performed using the toggleSearchAndResetButton function. It is also triggered when the user manually clicks the x button.

Todo: Improve styling @Badlapje do you have the energy to look into styling this button? I tried some myself, but it felt like going against the grain.

### CODE REVIEW

File: theme/base/stylesheets/pages/wayf.scss [hunk 1]
```diff
@@ -394,11 +394,13 @@
...
                 .search__field:hover + .search__submit,
-                .search__field:focus + .search__submit {
+                .search__field:focus + .search__submit,
+                .search__field:hover + .search__reset,
+                .search__field:focus + .search__reset {
                     @include button-reset($hoverBlue);
                 }
 
-                > .search__submit {
+                > .search__submit, .search__reset {
```
Comment (Badlapje) [inline theme/base/stylesheets/pages/wayf.scss:403]: nitpicky, but this selector should be on the next line (and possibly preceded by a >)

File: theme/cypress/integration/skeune/wayf/WayfGeneralBehaviour.js [hunk 2]
```diff
@@ -122,6 +128,14 @@ context('WAYF behaviour not tied to mouse / keyboard navigation', () => {
       cy.get('.wayf__remainingIdps .wayf__idp')
         .should('have.length', 5);
     });
+
+    it('Should reset the search text when clicking the reset button', () => {
```
Comment (Badlapje) [inline theme/cypress/integration/skeune/wayf/WayfGeneralBehaviour.js:132]: 💯 for tests :-)

**Review by Badlapje — APPROVED** [review]
Looks good to me.  Styling is off indeed, i'll make a PR on this branch shortly

**Comment by MKodde** [comment]
Thanks a million for helping me out on this one! One thing that might need attention is that on initial focus the clear button is shown before text is entered into the search field. This can/should be addressed in another PR IMHO.

**Pull request merged**